tests/batch_test.go

```diff
@@ -1,106 +1,129 @@
+	client := c.TestClient()
+	assert := c.Assert()
 
-	require.Len(batch.Shipments, 1)
-	assert.Equal("postage_purchased", batch.Shipments[0].BatchStatus)
+	batch, _ := client.CreateAndBuyBatch(c.fixture.OneCallBuyShipment(), c.fixture.OneCallBuyShipment())
 
-	shipment, err := client.GetShipment(batch.Shipments[0].ID)
-	require.NoError(err)
+	assert.Equal(reflect.TypeOf(&easypost.Batch{}), reflect.TypeOf(batch))
+	assert.True(strings.HasPrefix(batch.ID, "batch_"))
+	assert.Equal(2, batch.NumShipments)
+}
 
-	if addr := shipment.BuyerAddress; assert.NotNil(addr) {
-		assert.Equal("Burbank", addr.City)
-		assert.Equal("US", addr.Country)
-		assert.Equal("Bugs Bunny", addr.Name)
-		assert.Equal("8185551212", addr.Phone)
-		assert.Equal("4000 Warner Blvd", addr.Street1)
-	}
+func (c *ClientTests) TestBatchBuy() {
+	client := c.TestClient()
+	assert := c.Assert()
 
-	// Assert on fees.
-	if fees := shipment.Fees; assert.GreaterOrEqual(len(fees), 3) {
-		assert.Equal("0.00000", fees[0].Amount)
-		assert.Equal("7.92000", fees[1].Amount)
-		assert.Equal("1.00000", fees[2].Amount)
-	}
+	batch, _ := client.CreateBatch(
+		c.fixture.OneCallBuyShipment(),
+		nil,
+	)
 
-	// Assert on parcel.
-	if parcel := shipment.Parcel; assert.NotNil(parcel) {
-		assert.Equal("RegionalRateBoxA", parcel.PredefinedPackage)
-		assert.Equal(float64(64), parcel.Weight)
-	}
+	boughtBatch, _ := client.BuyBatch(batch.ID)
 
-	// Assert on tracker.
-	if tracker := shipment.Tracker; assert.NotNil(tracker) {
-		assert.NotEmpty(tracker.TrackingCode)
-		assert.NotEmpty(tracker.ShipmentID)
-	}
+	assert.Equal(reflect.TypeOf(&easypost.Batch{}), reflect.TypeOf(boughtBatch))
+	assert.Equal(1, boughtBatch.NumShipments)
+}
+
+func (c *ClientTests) TestBatchCreateScanForm() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	batch, _ := client.CreateBatch(
+		c.fixture.OneCallBuyShipment(),
+		nil,
+	)
+
+	boughtBatch, _ := client.BuyBatch(batch.ID)
+
+	batchWithScanForm, _ := client.CreateBatchScanForms(boughtBatch.ID, "")
+
+	assert.Equal(reflect.TypeOf(&easypost.Batch{}), reflect.TypeOf(batchWithScanForm))
+}
+
+func (c *ClientTests) TestBatchAddRemoveShipment() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	shipment, _ := client.CreateShipment(c.fixture.OneCallBuyShipment())
+
+	batch, _ := client.CreateBatch()
+
+	batchWithShipment, _ := client.AddShipmentsToBatch(batch.ID, shipment)
+
+	assert.Equal(1, batchWithShipment.NumShipments)
+
+	batchWithoutShipment, _ := client.RemoveShipmentsFromBatch(batch.ID, shipment)
+
+	assert.Equal(0, batchWithoutShipment.NumShipments)
+}
+
+func (c *ClientTests) TestBatchLabel() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	batch, _ := client.CreateBatch(
+		c.fixture.OneCallBuyShipment(),
+		nil,
+	)
+
+	boughtBatch, _ := client.BuyBatch(batch.ID)
+
+	batchWithLabel, _ := client.GetBatchLabels(boughtBatch.ID, "ZPL")
+
+	assert.Equal(reflect.TypeOf(&easypost.Batch{}), reflect.TypeOf(batchWithLabel))
```

Comment:
Please make sure you're including all the original comments from the other test suites:

`We can't assert anything meaningful here because the scanform gets queued for generation and may not be immediately available`